modules/core/src/main/java/org/apache/ignite/internal/util/ArrayUtils.java

```diff
@@ -296,6 +299,21 @@ public static boolean nullOrEmpty(boolean[] arr) {
         return nullOrEmpty(vals) ? Collections.emptyList() : Arrays.asList(vals);
     }
 
+    /**
+     * Converts array to {@link Set}.
+     *
+     * <p>Note that unlike {@link Arrays#asList(Object[])}, this method is {@code null}-safe. If {@code null} is passed in, then empty set
+     * will be returned.
+     *
+     * @param vals Array of values
+     * @param <T>  Array type.
+     * @return {@link Set} instance for input array.
+     */
+    @SafeVarargs
+    public static <T> Set<T> asSet(@Nullable T... vals) {
+        return nullOrEmpty(vals) ? Collections.emptySet() : Stream.of(vals).collect(Collectors.toSet());
```

Comment:
`Collections.emptySet()` return an immutable set, whereas `Stream.of(vals).collect(Collectors.toSet())` returns mutable. To me it would be better to keep semantic consistent across the whole range of arguments. Thus I propose to use `Set.of(...)` in case of non empty array.

BTW the very same issue exists for `asList` method. Could you please fix this as well?

Reply:
this one is not resolved actually

Reply:
already rewritten:
    public static <T> Set<T> asSet(@Nullable T... vals) {
        if (nullOrEmpty(vals)) {
            return Collections.emptySet();
        } else {
            HashSet<T> set = new HashSet<>(vals.length);
            Collections.addAll(set, vals);
            return set;
        }
    }

Reply:
and how does it fix mentioned concern?

Reply:
got it.